### PR TITLE
Only send deployment events to Kafka not ZK

### PR DESCRIPTION
**Pull request by davidxia**
We don't need to send deployment group events to ZooKeeper.
Just send them to Kafka fire-and-forget style because this
data isn't critical. This vastly simplifies things.

### CODE REVIEW

**Comment by davidxia** [comment]
@rohansingh

File: helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java [hunk 1]
```diff
@@ -597,9 +597,7 @@ public void rollingUpdateStep(final DeploymentGroup deploymentGroup,
           "rolling-update on deployment-group " + deploymentGroup.getName() + " failed", e);
     }
 
-    if (deploymentGroupHistoryWriter != null) {
-      deploymentGroupHistoryWriter.saveHistoryItems(opsEvents.getEvents());
-    }
+    kafkaSender.send((KafkaRecord) opsEvents.getEvents());
```
Comment (rohansingh) [inline helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java]: Pseudo-code:
```
for (final X event : opsEvents.getEvents()) {
  kafkaSender.send(KafkaRecord.of(event.key, event.timestamp, event.toBytes()));
}
```

File: helios-services/src/main/java/com/spotify/helios/servicescommon/KafkaSender.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+/**
+ * A class that wraps {@link org.apache.kafka.clients.producer.KafkaProducer}.
+ */
+public class KafkaSender {
+
+  private static final Logger log = LoggerFactory.getLogger(KafkaSender.class);
+
+  private final Optional<KafkaProducer<String, byte[]>> kafkaProducer;
+
+  public KafkaSender(final Optional<KafkaProducer<String, byte[]>> kafkaProducer) {
```
Comment (gimaker) [inline helios-services/src/main/java/com/spotify/helios/servicescommon/KafkaSender.java]: why optional?

Comment (gimaker) [inline helios-services/src/main/java/com/spotify/helios/servicescommon/KafkaSender.java]: i see

**Comment by gimaker** [comment]
Maybe pass an interface to ZKMasterModel to make it less coupled?

E.g. interface DeploymentEventSink { void emitEvent(DeploymentEvent ev); }

File: helios-services/src/main/java/com/spotify/helios/servicescommon/KafkaSender.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    this.kafkaProducer = kafkaProducer;
+  }
+
+  public void send(final KafkaRecord record) {
+    if (kafkaProducer.isPresent()) {
+      try {
+        final Future<RecordMetadata> future = kafkaProducer.get().send(
+            new ProducerRecord<String, byte[]>(record.getKafkaTopic(), record.getKafkaData()));
+        final RecordMetadata metadata = future.get(5, TimeUnit.SECONDS);
```
Comment (rohansingh) [inline helios-services/src/main/java/com/spotify/helios/servicescommon/KafkaSender.java]: I would s/5/KAFKA_SEND_TIMEOUT/ or something similar, so it's not just a magic number in the middle of the class.

**Comment by rohansingh** [comment]
This is sweet. Could we rip out the other Kafka stuff and use this for sending task history as well?

**Comment by rohansingh** [comment]
Never mind, just saw #567 :)

:+1:

**Comment by rohansingh** [comment]
:+1: 

:shipit:

**Pull request merged**